### PR TITLE
Adding staintools

### PR DESCRIPTION
**Pull request by trislaz**

### CODE REVIEW

File: runtime/environment-gpu.yml [hunk 1]
```diff
@@ -43,3 +43,6 @@ dependencies:
   - openslide-python==1.2.0
   - timm==0.4.12
   - opencv=4.7.0
+  - pip
```
Comment (ejm714) [inline runtime/environment-gpu.yml:46]: we shouldn't need this line

**Review by ejm714 — CHANGES_REQUESTED** [review]
Based on the [README](https://github.com/Peter554/StainTools) for `staintools`, this requires another package called `SPASM`. Can you add that to the requirements as well?

File: runtime/environment-gpu.yml (hunk 1 above)
Comment (trislaz) [reply runtime/environment-gpu.yml]: ok thank you i just erased it

File: runtime/environment-cpu.yml [hunk 2]
```diff
@@ -42,3 +42,6 @@ dependencies:
   - openslide-python==1.2.0
   - timm==0.4.12
   - opencv=4.7.0
+  - pip:
+    - staintools==2.1.2
+    - spams==2.6.5.4
```
Comment (ejm714) [inline runtime/environment-cpu.yml:47]: The staintools readme recommends conda instead of pip so let's move this outside of the pip block

**Review by ejm714 — CHANGES_REQUESTED** [review]
Let's install spams through [conda](https://anaconda.org/conda-forge/python-spams) per staintools install instructions

File: runtime/environment-cpu.yml (hunk 2 above)
Comment (trislaz) [reply runtime/environment-cpu.yml:47]: The conda repo has not been updated for three years, and this old version does not seem to be compatible with the current image. Isnt it possible to install it through PyPi ?

**Review by ejm714 — APPROVED** [review]

**Comment by trislaz** [comment]
Ok, thanks !

**Pull request merged**